File: gpt-3.5-ft-large/template_context_full/serial_construct_vector_length_clause.f90
#ifndef T1
!T1:serial construct vector_length clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        REAL(8), DIMENSION(10*LOOPCOUNT) :: a, b, c
        INTEGER :: x
        
        SEEDDIM(1) = 1
        #ifdef _OPENACCpragma acc serial
        DO x = 1, 10 * LOOPCOUNT
          a(x) = RAND(SEED)
          b(x) = RAND(SEED)
        END DO
        #endif
        #Loop testing can't be done
        #Loop testing can't be done.
        c(1:10*LOOPCOUNT) = a(1:10*LOOPCOUNT) + b(1:10*LOOPCOUNT)
        DO x = 1, 10*LOOPCOUNT
          IF (abs(c(x) - (a(x) + b(x))) .gt. 3) THEN
            errors = errors + 1
          END IF
  END DO
        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: gpt-4/stage2/serial_construct_vector_length_clause.cpp
#include "acc_testsuite.h"
#include <openacc.h>

#define N 1024
#define VECTOR_LENGTH 32

#ifndef T1
/*T1:serial construct vector_length clause,V:2.0-2.7*/
int test1(){
    int err = 0;
    srand(SEED);

    // Initialize host array
    float* array_host = new float[N];
    for(int i=0;i<N;i++){
        array_host[i] = rand() % 100;
    }

    // Compute on device
    float* array_device = new float[N];
    #pragma acc data copyin(array_host[0:N]) copy(array_device[0:N])
    {
        #pragma acc serial vector_length(VECTOR_LENGTH)
        for(int i=0; i<N; i++){
           array_device[i] = array_host[i];
        }
    }

    // Compare host and device result
    for(int i=0;i<N;i++){
        if(array_host[i] != array_device[i]){
            err = 1;
            break;
        }
    }

    delete[] array_host;
    delete[] array_device;

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}